include/envoy/router/router.h

```diff
@@ -316,8 +316,10 @@ class RouteEntry {
    * adding or removing headers. This should only be called ONCE immediately after receiving an
    * upstream's headers.
    * @param headers supplies the response headers, which may be modified during this call.
+   * @param request_info holds additional information about the request.
    */
-  virtual void finalizeResponseHeaders(Http::HeaderMap& headers) const PURE;
+  virtual void finalizeResponseHeaders(Http::HeaderMap& headers,
+                                       const AccessLog::RequestInfo& request_info) const PURE;
```

Comment:
I think that RequestInfo should be moved out of the AccessLog namespace because it's no longer only used for logging.

Reply:
#2163